drivers/storage/portworx/portworx.go

```diff
@@ -1287,6 +1286,46 @@ func setTLSDefaults(
 	}
 }
 
+func setTelemetryDefaults(
+	toUpdate *corev1.StorageCluster,
+	pxVersion *version.Version,
+) error {
+	// Disable telemetry if it's not supported but enabled explicitly
+	if pxutil.IsTelemetryEnabled(toUpdate.Spec) && pxVersion.LessThan(pxutil.MinimumPxVersionCCM) {
```

Comment:
yes, this block is checking if user is enabling telemetry but px version is too low. with the first part, we only disable it when user enables it, otherwise this telemetry section should be nil and won't show in the stc spec.